HeroVault/HeroVault/Features/Home/HomeScreen/HomeScreenViewModel.swift

```diff
@@ -0,0 +1,136 @@
+    private weak var delegate: ViewModelProtocol?
+    
+    init(homeScreenRepository: HomeScreenRepositoryType, delegate: ViewModelProtocol) {
+        self.homeScreenRepository = homeScreenRepository
+        self.delegate = delegate
+        self.marvelDataType = .character
+        isSearching = false
+        hideNoResultsText = true
+    }
+    
+    // MARK: Computes properties
+    
+    var marvelDataCount: Int {
+        marvelData.count
+    }
+    
+    var filteredMarvelDataCount: Int {
+        filteredMarvelData.count
+    }
+    
+    var numberOfSections: Int {
+        isSearching ? filteredMarvelDataCount : marvelDataCount
+    }
+    
+    // MARK: Functions
+    
+    func set(marvelDataType: EntityType) {
+        self.marvelDataType = marvelDataType
+    }
+    
+    func fetchMarvelNameAndImage(for marvelIndex: Int) -> (String, String) {
+        isSearching ?
+        (filteredMarvelData[marvelIndex].name, "\(filteredMarvelData[marvelIndex].thumbnail)/portrait_incredible.jpg".convertToHttps()) :
+        (marvelData[marvelIndex].name, "\(marvelData[marvelIndex].thumbnail)/portrait_incredible.jpg".convertToHttps())
+    }
+    
+    func fetchMarvelData(atIndex: Int) -> MarvelData? {
+        isSearching ? filteredMarvelData[atIndex] : marvelData[atIndex]
+    }
+    
+    func fetchMarvelData() {
+        marvelDataType == .character ? fetchCharacters() : fetchComics()
+    }
+    
+    func filterMarvelData(filteredText: String) {
+        
+        isSearching = true
+        guard !filteredText.isEmpty else {
+            isSearching = false
+            hideNoResultsText = true
+            return
+        }
+        
+        filteredMarvelData = marvelData.filter { marvelItem in
+            marvelItem.name.lowercased().contains(filteredText.lowercased())
+        }
+        
+        loadSearchNotFoundText()
+        delegate?.reloadView()
+    }
+    
+    func handleSegmentedControl(segmentedControlTitle: String) {
+        isSearching = false
+        segmentedControlTitle == "Characters" ? set(marvelDataType: EntityType.character) : set(marvelDataType: EntityType.comic)
+        fetchMarvelData()
+    }
+    
+    private func fetchCharacters() {
+        marvelData = []
+        homeScreenRepository?.fetchCharacters { [weak self] result in
+            guard let self else { return }
+            switch result {
+            case .success(let characters):
+                for character in characters.data.results {
+                    if !character.name.isEmpty && !character.overview.isEmpty && !checkIfImageIsAvailable(thumbnail: character.thumbnail) {
+                        marvelData.append(character)
+                    }
+                }
+                delegate?.reloadView()
+            case .failure(let error):
+                self.error = error
+            }
+        }
+    }
+    
+    private func fetchComics() {
+        marvelData = []
```

Comment:
Why do you need this? 
```suggestion
        marvelData = []
```

Reply:
it looks like it resets the list because the list looks like it's storing comings or chararecters so when the seg control is clicked, it empties the list from characters to comixs